### ISSUE TITLE
Tab completion doesn't work when a completion contains a space

### ISSUE THREAD

**Issue by sam-barr** [xmonad/xmonad-contrib#455]
### Problem Description

When using XMonad.Prompt, if you tab-complete past a completion which contains a space, instead of moving to the next completion it instead deletes the last work currently in the prompt and adds the text of the currently highlighted completion to the prompt.

### Configuration File

Here is a minimal prompt which recreates the issue (the completion will get stuck on `"c d"`), along with the XPConfig I am using.

```haskell
data Test1 = Test1

instance XPrompt Test1 where
    showXPrompt Test1 = "Test1: "
    commandToComplete Test1 s = s
    nextCompletion Test1 = getNextCompletion

test1Prompt :: XPConfig -> X ()
test1Prompt conf = mkXPrompt Test1 conf (const $ return ["a", "b", "c d", "e"]) $ const (return ())

myXPConfig :: XPConfig
myXPConfig = def
    { font = "xft:Source Code Pro:dpi=336:size=8:style=bold"
    , promptBorderWidth = 6
    , height = 100
    , position = Top
    , alwaysHighlight = True
    , maxComplRows = Just 1
    , historySize = 0
    }

```

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my configuration with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

**Comment by slotThe** [comment]
This is due to `XMonad.Prompt.hlComplete` (local function of `handleCompletion`, introduced in 526336ecec98aa2fe1cc98521473372b0664e19d) not functioning *quite* correctly.  

Do you want to look into this yourself?  If not, I'll _try_ to find the time

**Comment by slotThe** [comment]
@sam-barr in case you missed it: A potential fix is now implemented in #459.  Can you confirm that this fixes the issue for you?
